docs-api/documentation.md

```diff
@@ -0,0 +1,370 @@
+coremltools API v4.1 Documentation
+==================================
+
+This document describes the following:
+
+* [Viewing documentation](#viewing-documentation)
+* [Updating the documentation](#updating-the-documentation)
+* [Generating a new API reference](#generating-a-new-api-reference)
+
+
+## Viewing documentation {#viewing-documentation}
+
+For coremltools documentation, see the following:
+
+* Core ML Tools [README](README.md) file for this repository
+* [Guides and examples](https://coremltools.readme.io/) with installation and troubleshooting
+* [API Reference](https://apple.github.io/coremltools/index.html)
+* [Building from Source](https://github.com/apple/coremltools/blob/master/BUILDING.md)
+* [Release Notes](https://github.com/apple/coremltools/releases/) for the current release and previous releases
+* [Previous coremltools API (v3.4)](https://apple.github.io/coremltools/docs-api-version3/index.html)
```

Comment:
This URL doesn't work for me.

I like the idea of having previous version of the API online. However I think we need to automate this process. Having version numbers hardcoded is brittle. It's too easy for someone to forget to change them before releasing the next version.

Reply:
Agreed, removed. I think a link to the previous version should be in the release notes for that version, which will never change. So, to be specific, I would put the link for "coremltools API (v3.4) reference" in the text for the [coremltools 3.4](https://github.com/apple/coremltools/releases/tag/3.4) page.